### ISSUE TITLE
dyld[38703]: Library not loaded: @rpath/FBSDKCoreKit.framework/FBSDKCoreKit

### ISSUE THREAD

**Issue by salahelsayed99** [facebook/facebook-ios-sdk#2121]
### Checklist before submitting a bug report

- [X] I've updated to the latest released version of the SDK
- [X] I've searched for existing [GitHub issues](https://github.com/facebook/facebook-ios-sdk/issues)
- [X] I've looked for existing answers on [Stack Overflow](https://facebook.stackoverflow.com), the [Facebook Developer Community Forum](https://developers.facebook.com/community/) and the [Facebook Developers Group](https://www.facebook.com/groups/fbdevelopers)
- [X] I've read the [Code of Conduct](https://github.com/facebook/facebook-ios-sdk/blob/main/CODE_OF_CONDUCT.md)
- [X] This issue is not security related and can safely be disclosed publicly on GitHub

### Xcode version

Version 13.1 

### Facebook iOS SDK version

15.0.0 

### Dependency Manager

CocoaPods

### SDK Framework

Core

### Goals

Avoid crash the app when i use the "15.0.0" version of sdk, if i used an oldest version everything is fine.

### Expected results

Avoid crash 

### Actual results

App crashes in the beginning when i used "15.0.0" version of sdk, if i used an oldest version everything is fine.

I tried to use "use_modular_headers!".
I tried to delete derived data.

**Comment by tallot13** [comment]
same

**Comment by abbassk** [comment]
same here

**Comment by cristian1980** [comment]
I encountered the same issue. Are you guys using obj-C/C++ ?

I was hoping this was the issue for me. On the mainpage of this GIT there is a warning that I should start writing some wrappers around FB code.

> ⚠️ Be Advised ⚠️
> --
> We have begun rewriting the iOS SDK in Swift in order to modernize the code base.Please monitor the changelog for updates to existing interfaces but keep in mind that some interfaces will be unstable during this process. As such, updating to a minor version may introduce compilation issues related to language interoperability. Using symbols now defined in Swift may require using @import syntax from Objective-C and using C++ will likely require workarounds like creating wrappers in Objective-C.Please bear with us as we work towards providing an improved experience for integrating with the Facebook platform.
>

**Comment by tallot13** [comment]
i use swift

**Comment by christinachanhk** [comment]
Will that be the reason on old Chips? 

As I got 2 Macs, the one with M1 Pro core works fine, but the other with old core encounter the same error.

**Comment by salahelsayed99** [comment]
> Will that be the reason on old cores?
> 
> As I got 2 Macs, the one with M1 Pro core works fine, but the other with old core encounter the same error.
> 
> The Mac with old core: Mac mini (2018) Processor 3.2 GHz 6-Core Intel Core i7

My mac: Mac pro (2014 mid) Processor 2.5 GHz Quad-Core Intel Core i7

**Comment by astrokin** [comment]
the same for me. adhoc works, but release config crash on start with 
```
DYLD error: Library not loaded: @rpath/FBSDKCoreKit.framework/FBSDKCoreKit + 0
```

**Comment by Pashkagreen** [comment]
The same, tried to change different rpaths, nothing helped

**Comment by wangjian201314** [comment]
<img width="1050" alt="image" src="https://user-images.githubusercontent.com/16216900/194989385-c00005d8-babe-4ef3-b616-dfa75647017f.png">
it is ok ！

**Comment by astrokin** [comment]
downgrade to `14.1.0` helped

**Comment by Pashkagreen** [comment]
> downgrade to `14.1.0` helped

+1

**Comment by clding** [comment]
same +1
xcode13.4.1 + swift + fb15.0.0

**Comment by lidorf-gurushots** [comment]
same error
Xcode 14.0.1
SDK version: 15.0.0
any news about this error?

**Comment by DIGITALSQUAD** [comment]
> <img alt="image" width="1050" src="https://user-images.githubusercontent.com/16216900/194989385-c00005d8-babe-4ef3-b616-dfa75647017f.png">
> 
> In your Xcode ,go the General tab of your Main Project.. there you will find an Embedded Binaries field and add the required framework there . Crash will be resolved.

Additionally added FBSDKGamingServicesKit.xcframework fixed this issue.

**Comment by liweican1992** [comment]
same here
down to 14.1.0 helped

**Comment by 1074332133** [comment]
> 
in my xcode   add framework will cause build failed
duplicate output file  on task: ProcessXCFramework 
Multiple commands produce

**Comment by joesus** [comment]
[salahelsayed99](https://github.com/salahelsayed99) thanks for opening this issue! Can you provide a sample project or Podfile to help reproduce?

**Comment by umair-khan97** [comment]
Try Installing with SPM

**Comment by djMax** [comment]
what's SPM?

**Comment by umair-khan97** [comment]
> what's SPM?

Swift Package Manager

**Comment by djMax** [comment]
Interesting. I'm in a react-native env, via react-native-fbsdk-next, so I don't think that's an option, but it certainly sounds like a plausible explanation as to why it's breaking.

**Comment by Achilles0509** [comment]
> 

I have 2 Macs, but error only happens on old mac.

<img width="572" alt="image" src="https://user-images.githubusercontent.com/79822851/198114991-402e41f7-14a6-4036-935e-4c82333d258f.png">

Here is a detailed error message.
```
Exception Type:  EXC_CRASH (SIGABRT)
Exception Codes: 0x0000000000000000, 0x0000000000000000
Exception Note:  EXC_CORPSE_NOTIFY
Termination Reason: DYLD 1 Library missing
Library not loaded: @rpath/FBSDKCoreKit.framework/FBSDKCoreKit
Referenced from: /Users/USER/Library/Developer/CoreSimulator/Devices/5FB219D0-6B8C-4EFC-8B2D-5BB8D9573900/data/Containers/Bundle/Application/A89C2832-022C-43C5-BB30-107DE9587660/Awesome.app/Awesome
Reason: tried: '/Library/Developer/CoreSimulator/Profiles/Runtimes/iOS 15.4.simruntime/Contents/Resources/RuntimeRoot/usr/lib/swift/FBSDKCoreKit.framework/FBSDKCoreKit' (no such file), '/usr/lib/swift/FBSDKCoreKit.framework/FBSDKCoreKit' (no such file), '/Library/Developer/CoreSimulator/Profiles/Runtimes/iOS 15.4.simruntime/Contents/Resources/RuntimeRoot/usr/lib/swift/FBSDKCoreKit.framework/FBSDKCoreKit' (no such file), '/usr/lib/swift/FBSDKCoreKit.framework/FBSDKCoreKit' (no such file), '/Users/mactron/Library/Developer/CoreSimulator/Devices/5FB219D0-6B8C-4EFC-8B2D-5BB8D9573900/data/Containers/Bundle/Application/A89C2832-022C-43C5-BB30-107DE9587660/Awesome.app/Frameworks/FBSDKCoreKit.framework/FBSDKCoreKit' (no such file), '/Users/mactron/Libra
(terminated at launch; ignore backtrace)
```

**Comment by Achilles0509** [comment]
> > <img alt="image" width="1050" src="https://user-images.githubusercontent.com/16216900/194989385-c00005d8-babe-4ef3-b616-dfa75647017f.png">
> > In your Xcode ,go the General tab of your Main Project.. there you will find an Embedded Binaries field and add the required framework there . Crash will be resolved.
> 
> Additionally added FBSDKGamingServicesKit.xcframework fixed this issue. ![スクリーンショット 2022-10-13 17 06 03](https://user-images.githubusercontent.com/6149827/195539407-e889bf85-00af-4276-9c38-41b7176375cc.png)

It worked for me, but I don't like this approach.
It changes `project.pbxproj` file, but I don't want to merge changes to main branch.
Does anyone have better solution?

**Comment by khoogheem** [comment]
Same thing on 15.1.0 using pods

**Comment by MrAbraham98** [comment]
> <img alt="image" width="1050" src="https://user-images.githubusercontent.com/16216900/194989385-c00005d8-babe-4ef3-b616-dfa75647017f.png">
> 
> In your Xcode ,go the General tab of your Main Project.. there you will find an Embedded Binaries field and add the required framework there . Crash will be resolved.

its work for me thnx. my problem is project created m1 arch and i am working  on intel arch it seems arch doing conflict.

**Comment by Achilles0509** [comment]
> > <img alt="image" width="1050" src="https://user-images.githubusercontent.com/16216900/194989385-c00005d8-babe-4ef3-b616-dfa75647017f.png">
> > In your Xcode ,go the General tab of your Main Project.. there you will find an Embedded Binaries field and add the required framework there . Crash will be resolved.
> 
> its worked for me thnx. my problem is project created m1 arch and i am working on intel arch it seems arch doing conflict.

Yes, it works, but as I said [here](https://github.com/facebook/facebook-ios-sdk/issues/2121#issuecomment-1292524229) it's not recommended way.
Do you have better solution?

**Comment by foermo4-hao** [comment]
> > > <img alt="image" width="1050" src="https://user-images.githubusercontent.com/16216900/194989385-c00005d8-babe-4ef3-b616-dfa75647017f.png">
> > > In your Xcode ,go the General tab of your Main Project.. there you will find an Embedded Binaries field and add the required framework there . Crash will be resolved.
> > 
> > 
> > its worked for me thnx. my problem is project created m1 arch and i am working on intel arch it seems arch doing conflict.
> 
> Yes, it works, but as I said [here](https://github.com/facebook/facebook-ios-sdk/issues/2121#issuecomment-1292524229) it's not recommended way. Do you have better solution?

Don't want to merge to main too

**Comment by salahelsayed99** [comment]
> [salahelsayed99](https://github.com/salahelsayed99) thanks for opening this issue! Can you provide a sample project or Podfile to help reproduce?

I am sorry for late reply unfortunately i can't reproduce it but it obviously happened in the newest version but downgrade to 14.1.0 helped

**Comment by 2sem** [comment]
pod 'FBSDKCoreKit', '14.1.0' works
target: iOS 13
xcode: 14.1

**Comment by kidc2458** [comment]
I found that the content of ios/pods/Target Support Files/Pods-{project name}/Pods-{project name}-framework.sh which was created after running 'pod install' was wrong.
I have two mac os that build the same project, and both mac os have the same environment. (maybe)
xcode 14.1
mac os 13.0
node 18.7.0
ruby 2.6.10
... Other than that, the project settings are all the same.

In the build of Mac OS that has the same problem as this issue, the 'install_framework' entry in 'Pods-{project name}-framework.sh' contains the path of FBSDKCorekit.framework It hasn't been done.

All project settings on both Mac OS are the same. Since I use SVN, I have removed all unnecessary files in the project with svn revert, svn cleanup, etc.

But the result of 'pod install' is different.

I'm looking for something that affects 'pod install' among other environment settings.

Please let me know if anything comes to mind.

**Comment by kidc2458** [comment]
I haven't been able to find the exact cause of this issue in the end.
I set up all the environments to build the project from scratch, and in the end I was able to get the right result.

I installed and built the same project on a total of 4 Mac with different specs and it worked on the Mac I mentioned before.

I uninstalled brew, node, cocoapods, ruby, xcode, yarn respectively and reinstalled it.

**Comment by yhoseph** [comment]
I fixed this issue by running this command "pod install --repo-update"

**Comment by zeucxb** [comment]
![WhatsApp Image 2022-11-11 at 13 33 19](https://user-images.githubusercontent.com/11702749/201387132-db1983ff-7976-4d51-8f9e-51be91fdbf3b.jpeg)

I just add `FBSDKCoreKit` and `FBSDKCoreKit_Basics` and it works 🙌🏼

**Comment by sajibsaha4647** [comment]
dyld[24623]: Library not loaded: @rpath/FBSDKGamingServicesKit.framework/FBSDKGamingServicesKit .

still i have problem with this . nothing works for me .

**Comment by sajibsaha4647** [comment]
> ![WhatsApp Image 2022-11-11 at 13 33 19](https://user-images.githubusercontent.com/11702749/201387132-db1983ff-7976-4d51-8f9e-51be91fdbf3b.jpeg)
> 
> I just add `FBSDKCoreKit` and `FBSDKCoreKit_Basics` and it works 🙌🏼

in this way dyld[24623]: Library not loaded: @rpath/FBSDKGamingServicesKit.framework/FBSDKGamingServicesKit 
same i added and it works for me .

**Comment by triplef** [comment]
> I found that the content of ios/pods/Target Support Files/Pods-{project name}/Pods-{project name}-framework.sh which was created after running 'pod install' was wrong. 

I also believe that’s the cause: for some reason the script is missing the `install_framework` call after updating to 15.1.0:

![image](https://user-images.githubusercontent.com/6037/203798062-03384de0-8d27-4671-b744-210b6c9a24ac.png)

@joesus any idea what might be going on here?

**Comment by bgoncharuck** [comment]
Same

**Comment by bastien-shopopop** [comment]
Updating cocoapods from 1.11.2 to 1.11.3 fix the issue for me.

**Comment by bgoncharuck** [comment]
For Flutter users of flutter_facebook_auth.
Temporary fix:

In pubspec.yaml
```
  flutter_facebook_auth:
    git:
      url: https://github.com/darwin-morocho/flutter-facebook-auth.git
      ref: 1ba68fa7357075740591b31a511bb9ba67c8740c # pragma: allowlist secret
      path: facebook_auth
```

**Comment by mtallenca** [comment]
I had 3 macs with XCode 14.1 - all would build / install ipa, 2 of the builds would produce the FBSDKCoreKit error.  Running xcode-select --install fixed those 2.

**Comment by yaoxp** [comment]
Same.

**Comment by liushengxu** [comment]
> We agree with you. Is it resolved ？？Dear ！！

**Comment by bgoncharuck** [comment]
Any progress on fixing it?

**Comment by NamHofstadter** [comment]
> > 
> 
> **Embedded Binaries field and add the required framework not work** in my xcode will build failed info: duplicate output file on task: ProcessXCFramework Multiple commands produce

have you fix this？

**Comment by dragouf** [comment]
`Embedded Binaries field and add the required framework not work in my xcode will build failed info: duplicate output file on task: ProcessXCFramework Multiple commands produce`

same problem for me... I use flutter library named facebook_app_events. Here is the related ticket (https://github.com/oddbit/flutter_facebook_app_events/issues/279) but I still don't have a fix

**Comment by NamHofstadter** [comment]
> > Embedded Binaries field and add the required framework not work in my xcode will build failed info: duplicate output file on task: ProcessXCFramework Multiple commands produce
> 
> same problem for me... I use flutter library named facebook_app_events. Here is the related ticket ([oddbit/flutter_facebook_app_events#279](https://github.com/oddbit/flutter_facebook_app_events/issues/279)) but I still don't have a fix

I use facebook_app_events too! And also 'flutter_login_facebook' library
I have fix it by add just FBSDKCoreKit.xcframework
![image](https://user-images.githubusercontent.com/26591903/207012289-9904b888-609f-4739-a7a1-b06a25f4b3e2.png)
hope that will help u
In other Mac,the 'facebook_app_events' library works just fine without doing anything.
But in my Mac,I have to add FBSDKCoreKit.xcframework
that's weird,same Mac OS version,same XCode version,same Intel Core.
nothing is different!
If u find something,Please let me Know

**Comment by felipe-xavier** [comment]
> <img alt="image" width="1050" src="https://user-images.githubusercontent.com/16216900/194989385-c00005d8-babe-4ef3-b616-dfa75647017f.png">
> 
> In your Xcode ,go the General tab of your Main Project.. there you will find an Embedded Binaries field and add the required framework there . Crash will be resolved.

This and downgrading are just workarounds, my team has M1 and intel machines which get conflicts when including the SDK manually solution and we'd rather not downgrade the SDK.

M1 machines work fine with SDK 15, but when we add the SDK manually to make intel machines work, it just breaks the M1 machines later due to duplicate entries.

Please, update the SDK with a proper fix.

**Comment by greymag** [comment]
_@joesus is anyone working on a fix for this? This happens on Intel mac and this is pretty huge problem with current SDK version._

**Comment by liushengxu** [comment]
is anyone working on a fix for this? This happens on Intel mac and this is pretty huge problem with current SDK version.

**Comment by felipe-xavier** [comment]
is anyone working on a fix for this SDK? This happens on Intel mac and this is pretty huge problem with current SDK version.

**Comment by jamesho86** [comment]
blocked by this issue for a while and still waiting for a proper fix

**Comment by mansilaad** [comment]
+1 Same issue

**Comment by katomizuki** [comment]
+1 Same Issue

**Comment by K9i-0** [comment]
+1 Same Issue

**Comment by niwatly** [comment]
+1 Same Issue

**Comment by Frezyx** [comment]
+1 Same Issue

**Comment by jml0123** [comment]
> ![WhatsApp Image 2022-11-11 at 13 33 19](https://user-images.githubusercontent.com/11702749/201387132-db1983ff-7976-4d51-8f9e-51be91fdbf3b.jpeg)
> 
> I just add `FBSDKCoreKit` and `FBSDKCoreKit_Basics` and it works 🙌🏼

I tried this and get an error below:

```Multiple commands produce '/Users/jsmglorenzo/Library/Developer/Xcode/DerivedData/Runner-feevmrclfhykhyhhhrdtlavvyvge/Build/Products/Debug-development-iphoneos/Runner.app/Frameworks/FBSDKCoreKit_Basics.framework'```

Anyone else face this?

**Comment by Definitely-Not-Vlad** [comment]
@jml0123 After adding these to `Frameworks, Libraries, and Embedded Content`, you may have to remove it from `Build Phases` -> `Link Binary With Libraries`, so it's only in the `Embed Frameworks` build phase.

Note that you may also have to add `FBSDKGamingServicesKit` as well, at least I had to. Try without it first.

This worked for some, but not all of my projects.

I would really love for this to be resolved ASAP because it's blocking a ridiculous number of people. I'm not sure how I can help, but if anyone points me in the right direction I'm more than happy to give it a go.

**Comment by jml0123** [comment]
@Definitely-Not-Vlad thanks for the tip! Unfortunately this alone didn't work for me. _What finally worked_ using the 15.1.0 version was also to remove `FBSDKCoreKit.framework` from the `[CP] Embed Pods Frameworks` -> `Output Files` under `Build Phases`

<img width="639" alt="Screen Shot 2022-12-22 at 11 21 32 AM" src="https://user-images.githubusercontent.com/29313716/209178581-0e53f756-d689-4f73-9c44-f0d54080a169.png">

<img width="544" alt="Screen Shot 2022-12-22 at 11 23 38 AM" src="https://user-images.githubusercontent.com/29313716/209178124-04fc8c02-f008-4450-b78e-eb1cc83aa562.png">

To recap in case someone is facing the same issue:
1. I added `FBSDKCoreKit_Basics` and `FBSDKCoreKit` to Frameworks, Libraries and Embedded Content in `General`
2. Removed `FBSDKCoreKit_Basics` and `FBSDKCoreKit` from `Build Phases` -> `Link Binary With Libraries`
3. Removed reference to `FBSDKCoreKit.framework` from `Build Phases` -> `[CP] Embed Pods Frameworks` -> `Output Files`

**Comment by ehdgns3136** [comment]
> To recap in case someone is facing the same issue:
> 
> 1. I added `FBSDKCoreKit_Basics` and `FBSDKCoreKit` to Frameworks, Libraries and Embedded Content in `General`
> 2. Removed `FBSDKCoreKit_Basics` and `FBSDKCoreKit` from `Build Phases` -> `Link Binary With Libraries`
> 3. Removed reference to `FBSDKCoreKit.framework` from `Build Phases` -> `[CP] Embed Pods Frameworks` -> `Output Files`


@jml0123 
Thanks! This is working for me.
I'll write down what i did to make it work.
I'm building iOS from Unity.

My environment:
- MacOS Monterey 12.6
- Unity 2020.3.40f1
- Facebook-unity-sdk 15.1.0
- XCode 14.1
- Pod 1.11.3  
  

Process:
1. Version Up XCode 13.3 -> 14.1
2. Open .xcworkspaces file, not .xcodeproj file
Or it could generate `module 'FBSDKCoreKit' not found` error
(This may only apply to Unity users)
3. Add from `General` -> `Frameworks, Libraries, and Embedded Content`
- `FBSDKCoreKit_Basics.xcframework`
- `FBSDKCoreKit.xcframework`
- `FBSDKGamingServicesKit.xcframework`
![Screen Shot 2022-12-23 at 6 07 22 PM](https://user-images.githubusercontent.com/26455200/209306803-9a80b244-a662-4009-b54f-9a8c740e807d.png)
4. Remove from `Build Phases` -> `Link Binary With Libraries`
- `FBSDKCoreKit_Basics.xcframework`
- `FBSDKCoreKit.xcframework`
- `FBSDKGamingServicesKit.xcframework`
5. Remove from `Build Phases` -> `[CP] Embed Pods Frameworks`
- `${TARGET_BUILD_DIR}/${FRAMEWORKS_FOLDER_PATH}/FBSDKCoreKit_Basics.framework`

**Comment by devibrahimkarahan** [comment]
> > ![WhatsApp Image 2022-11-11 at 13 33 19](https://user-images.githubusercontent.com/11702749/201387132-db1983ff-7976-4d51-8f9e-51be91fdbf3b.jpeg)
> > I just add `FBSDKCoreKit` and `FBSDKCoreKit_Basics` and it works 🙌🏼
> 
> I tried this and get an error below:
> 
> ```
> Multiple commands produce '/Users/jsmglorenzo/Library/Developer/Xcode/DerivedData/Runner-feevmrclfhykhyhhhrdtlavvyvge/Build/Products/Debug-development-iphoneos/Runner.app/Frameworks/FBSDKCoreKit_Basics.framework'
> ```
> 
> Anyone else face this?
> 
> _Edit: After downgrading to 14.1.0 like someone mentioned above, I am also getting an issue with not found symbols:_
> 
> ```
> dyld[52084]: Symbol not found: (_$s8FBAEMKit11AEMReporterCN)
>   Referenced from: '/private/var/containers/Bundle/Application/BBFBF781-7C77-43D5-BC32-3C0CB37D6FF1/Runner.app/Frameworks/FBSDKCoreKit.framework/FBSDKCoreKit'
>   Expected in: '/private/var/containers/Bundle/Application/BBFBF781-7C77-43D5-BC32-3C0CB37D6FF1/Runner.app/Frameworks/FBAEMKit.framework/FBAEMKit'
> ```
> 
> Using: _MacOs Monterey 12.3.1 Flutter 3.3.10 • channel stable Dart 2.18.6 • DevTools 2.15.0 Xcode 13.4.1_

Just add FBSDKCoreKit.xcframework, not _Basics

**Comment by cwheeler42** [comment]
M1 Mac, using FBSDKCoreKit (only for analytics) via cocoapods. I too had to add the xcframework explicitly to my main project/target frameworks and embed & sign it to get past the crash. Please fix this. libs included via cocoapods should be built into the pod framework and included that way.

**Comment by evilutioner** [comment]
The same issue I able to reproduce on my environment:
- FB 14.1.0 / 15.0.0 / 15.1.0 SDK
- intel based macbook only
- cocoapods dependency
- Xcode 14.2

**Comment by idish** [comment]
It is an official library maintained by Facebook, so ridiculous that it is not fixed by now..
Are we expected to downgrade the library version then?

**Comment by felipecastrosales** [comment]
news about this? Same issue here.

**Comment by derwaldgeist** [comment]
Same issue here.

I have severe problems to get the SDK work with Unity 2019.4.40. I was forced to upgrade, because targeting Android 12 is now required by Google, and this broke the older Facebook SDK version 8 I was using. Upgrading for Android was already a nightmare. Now that I finally got this working, I ran into the iOS crash bug described here. Facebook's QA team seems to have serious issues. And they don't even seem to care about error reports. We're actually thinking about removing Facebook login from our app altogether. This takes up way too much dev time.

**Comment by derwaldgeist** [comment]
> > To recap in case someone is facing the same issue:

Thanks for the writeup. However, these manual tasks are unbearable if you're building app version often. Did you ever manage to automate this via Unity build scripts?

**Comment by derwaldgeist** [comment]
I was finally able to create a build script that resolved the problem for me:

```cs
#if UNITY_IOS
using System.IO;
using UnityEditor;
using UnityEditor.Callbacks;
using UnityEditor.iOS.Xcode;
using UnityEditor.iOS.Xcode.Extensions;
using UnityEngine;
using System.Collections.Generic;
#endif

public class FixFacebookSDK15_1
{
#if UNITY_IOS

	[PostProcessBuild]
	public static void OnPostprocessBuild(BuildTarget target, string buildPath) {
		if (target != BuildTarget.iOS) return;
		Debug.Log("Applying Facebook SDK 15.1.0 workaround. Remove me once Facebook has patched this.");

		string pbxProjectPath = PBXProject.GetPBXProjectPath(buildPath);
		PBXProject proj = new PBXProject();
		proj.ReadFromFile(pbxProjectPath);
		string targetGuid = proj.GetUnityFrameworkTargetGuid();
		string mainTargetGuid = proj.GetUnityMainTargetGuid();

		List<string> frameworks = new List<string>();
		frameworks.Add("FBSDKCoreKit");
		frameworks.Add("FBSDKGamingServicesKit");

		foreach (string framework in frameworks) {
			var fileName = framework + ".xcframework";
			var src = Path.Combine("Pods", framework, "XCFrameworks", fileName);
			var fullSrc = Path.Combine(buildPath, src);
			var dest = Path.Combine("Frameworks", fileName);
			var findBefore = proj.FindFileGuidByProjectPath(src);
			var name = proj.AddFile(src, src);
			proj.AddFileToBuild(targetGuid, name);
			var findAfter = proj.FindFileGuidByProjectPath(src);
			proj.AddFileToEmbedFrameworks(mainTargetGuid, name);
		}
		proj.WriteToFile(pbxProjectPath);
	}

#endif
}
```

Not sure if this is a generic solution, but my app did not crash anymore. I only added two of the frameworks and was even able to use Facebook Login with this configuration. You might want to add more in your case if needed.

In case you are experiencing issues with other frameworks, you might want add a sequence number to the PostProcessBuild attribute, e.g. like this:

`[PostProcessBuild(1000)]`

to control when this script is being executed in the list of other post process build scripts.

**Comment by romas34** [comment]
> I was finally able to create a build script that resolved the problem for me on Unity 2019.4:

> Not sure if this is a generic solution, but my app did not crash anymore. I only added two of the frameworks and was even able to use Facebook Login with this configuration. You might want to add more in your case if needed.
> 
> In case you are experiencing issues with other frameworks, you might want add a sequence number to the PostProcessBuild attribute, e.g. like this:
> 
> `[PostProcessBuild(1000)]`
> 
> to control when this script is being executed in the list of other post process build scripts.
> 
> The method `AddFileToEmbedFrameworks` doesn't seem to be available in newer Unity versions, so it might have to be replaced something else.

I can't add it like this :( Unity 2022.1.22. Also i tried to do this:
```
string targetName = project.GetUnityMainTargetGuid(); 
project.AddFrameworkToProject(targetName, "FBSDKCoreKit.framework", false);
or
project.AddFrameworkToProject(targetName, "FBSDKCoreKit.xcframework", false); 
```

i just get error in xcode:
There is no XCFramework found at '/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS16.2.sdk/System/Library/Frameworks/FBSDKCoreKit.xcframework'.

for both frameworks.

**Comment by derwaldgeist** [comment]
Ah, sorry, I kept some debug stuff in accidently. Glad you corrected the code.

**Comment by derwaldgeist** [comment]
Are you sure your code actually works? IIRC, I had to add the frameworks to the UnityFrameworkTarget as well.

**Comment by romas34** [comment]
> Are you sure your code actually works? IIRC, I had to add the frameworks to the UnityFrameworkTarget as well.

Yes.This is the working code. I spent a couple of days trying to fix this issue so that the app doesn't crash.

**Comment by derwaldgeist** [comment]
Ok, I can confirm it works on both 2019.4.31 and 2021.3.16. And it's simpler.

Was there a particular reason why you added the (1000) attribute?

**Comment by derwaldgeist** [comment]
Just realized that on Unity 2021, you need to add more frameworks. Here's my updated code:

```cs
#if UNITY_IOS
using System.IO;
using UnityEditor;
using UnityEditor.Callbacks;
using UnityEditor.iOS.Xcode;
using UnityEditor.iOS.Xcode.Extensions;
using UnityEngine;
using System.Collections.Generic;
#endif

// https://github.com/facebook/facebook-ios-sdk/issues/2121
public class FixFacebookSDK15_1
{
#if UNITY_IOS

	[PostProcessBuild]
	public static void OnPostprocessBuild(BuildTarget target, string buildPath) {
		if (target != BuildTarget.iOS) return;
		Debug.Log("Applying Facebook SDK 15.1.0 workaround. Remove me once Facebook has patched this.");

		string pbxProjectPath = PBXProject.GetPBXProjectPath(buildPath);
		PBXProject proj = new PBXProject();
		proj.ReadFromFile(pbxProjectPath);
		string mainTargetGuid = proj.GetUnityMainTargetGuid();

		List<string> frameworks = new List<string>();
		frameworks.Add("FBSDKCoreKit");
		frameworks.Add("FBSDKGamingServicesKit");
		frameworks.Add("FBAEMKit");
		frameworks.Add("FBSDKCoreKit_Basics");
		frameworks.Add("FBSDKLoginKit");
		frameworks.Add("FBSDKShareKit");
		foreach (string framework in frameworks) {
			var fileName = framework + ".xcframework";
			var src = Path.Combine("Pods", framework, "XCFrameworks", fileName);
			var fileGuid = proj.AddFile(src, src);
			proj.AddFileToBuild(mainTargetGuid, fileGuid);
			proj.AddFileToEmbedFrameworks(mainTargetGuid, fileGuid);
		}
		proj.WriteToFile(pbxProjectPath);
	}

#endif
}
```

With this setup, Facebook login is working. Not sure if you need even more frameworks if you're using other stuff. Interestingly, not even FBSDKLoginKit was needed on Unity 2019.4, although login worked fine.

**Comment by romas34** [comment]
> Was there a particular reason why you added the (1000) attribute?

No :) just to make sure it executes last.

**Comment by adityawibisana** [comment]
On my end, I have to change podfile, and change 
FBSDKCoreKit to version 15

```
pod 'FBSDKCoreKit', '~> 15.1'
```

**Comment by derwaldgeist** [comment]
I thought I had fixed this using the script above. The builds ran through and everything worked fine on my iPhone.

But now I wanted to push the version the app store. When uploading the build to the store, I got an error message:

<img width="1000" alt="image" src="https://user-images.githubusercontent.com/3984535/222540972-957ae07d-0449-4f2f-a02a-2ca86c85ef58.png">

It goes on like this:

<img width="1000" alt="image" src="https://user-images.githubusercontent.com/3984535/222541104-42eeff6c-c953-418b-8525-8a13e5003741.png">

This is for Unity 2019.4. On 2021.3, I am getting an additional error messages, which is related to Unity's own Framework:

<img width="745" alt="image" src="https://user-images.githubusercontent.com/3984535/222541899-fc53574a-5c0c-4b5c-9e71-80e16e8c0628.png">

It tells me that UnityFramework.framework it has a disallowed file "Frameworks". WTF?!

Any idea what is going on here and why the frameworks are in the wrong folder?! This is really driving me nuts. We have an important release soon, and it bugs if you notice something like this when trying to push your stuff to the stores... I wish we never integrated that crappy Facebook stuff in the first place.

**Comment by derwaldgeist** [comment]
There is still two issues that require manual tweaks:

* Unity 2021.3 does not set the "signing team" on the UnityFramework target.
* "Always Embed Standard Swift Libraries" has to be "NO" on the UnityFramework

I'm now trying to find a way to automate these two steps.

**Comment by abdullah1826** [comment]
> On my end, I have to change podfile, and change FBSDKCoreKit to version 15
> 
> ```
> pod 'FBSDKCoreKit', '~> 15.1'
> ```

I had the same version but it was giving me error now trying to change to older version

**Comment by xinatcg** [comment]
Same issue on Unity Facebook SDK 15.1

**Comment by kirti-swiggy** [comment]
Got fixed after updating Cocoapod to 1.12.0. Adding FBSDKCoreKit_Basics.xcframework in `Frameworks, Libraries, and Embedded Content` is causing build failure in CI systems.

**Comment by hochungwong** [comment]
Got if fixed after upgrading Cocoapod to 1.12.0.

Xcode version: 14.3
MacOS Version: Ventura 13.3 (22E252)
Chip: Intel Core i&

**Comment by devibrahimkarahan** [comment]
As @hochungwong mentioned, when i upgrade cocoapods to 1.12.0 (mine was 1.11.3), its fixed.

- Xcode 14.2
- M1, Ventura 13.0
- Pods 1.12.0
- Flutter: 3.7.6
- No podfile changes, still using 
```
  use_frameworks!
  use_modular_headers!
```
- No addition to Frameworks, Libraries, and Embedded Content

**Comment by umairali435** [comment]
Any update on this issue ?

**Comment by vishna** [comment]
seems to be fixed after upgrading cocoapods to 1.12.1 (flutter+facebook_app_events), so can confirm what @hochungwong & @devibrahimkarahan wrote

**Comment by spo-mmorales** [comment]
> 

It works for me

**Comment by spo-mmorales** [comment]
> seems to be fixed after upgrading cocoapods to 1.12.1 (flutter+facebook_app_events), so can confirm what @hochungwong & @devibrahimkarahan wrote

It works for me

**Comment by joshuarendora1** [comment]
Hi Good Day before we begin can i ask a little favor to Subscribe these
Youtube chanel?

Please click Link Below. 👇👇


https://youtube.com/shorts/va3bZK4wiKU?feature=share



YOU'RE SUBSCRIPTION IS REALLY A BIG HELP TO THE CHARITY WE CHOSEN TO
RECEIVED 75% OF THE SALLARY WE MAKE IN THE FUTURE.

WE ARE EXPECTING THAT YOU ARE ALSO IN LINE AND 1 OF THOSE PEOPLE ALSO WHO
SUPPORT WHAT WE MAKE AND PLAN IN THE FUTURE IN HELPING OTHERS ♥️♥️♥️

THANK YOU SO MUCH ♥️

**Comment by aboutlan** [comment]
The same problem, Have any solution except  **Adding FBSDKCoreKit.xcframework in Frameworks, Libraries, and Embedded Content**

**Comment by ivory-code** [comment]
> It works for me.

Xcode version: 14.3 (14E222b)
MacOS: Ventura 13.3.1 (22E261)
Processor: Intel Core i9

**Comment by ivory-code** [comment]
> Got if fixed by upgrading Cocoapod from 1.11.2 to 1.12.0.

It works for me.

Xcode version: 14.3 (14E222b)
MacOS: Ventura 13.3.1 (22E261)
Processor: Intel Core i9

**Comment by mtr1012** [comment]
Same issue on MacOS Sonoma. Upgrading the cocoapods to the latest version didn't help. Terrible facebook. Your SDK give us so much tired. We need to upgrade it constantly just to use your sign in feature? and your sdk breaks our project. Wtf?

**Comment by Zhiyi-Zhang** [comment]
Hi! After reviewing this issue, we're closing it because this build/link issue is related to an **older SDK or toolchain version** that is no longer current.

**Specific reason:** Build configuration issue - dyld Library not loaded with CocoaPods and SDK 15.0.0, Xcode 13.1.

The current SDK is v18.x and supports recent Xcode versions. Many build and linking issues from older SDK versions have been resolved in subsequent releases.

If you're still experiencing build issues on the latest SDK version with a current Xcode, please open a new issue with:
- SDK version, Xcode version, iOS version
- Integration method (SPM, CocoaPods, Carthage)
- Complete build error output
- Your project configuration details

Thanks for understanding!
